code/modules/mob/living/carbon/carbon.dm

```diff
@@ -107,20 +107,22 @@
 /mob/living/carbon/proc/toggle_throw_mode()
 	if(stat)
 		return
-	if(in_throw_mode)
-		throw_mode_off()
+	if(throw_mode)
+		throw_mode_off(THROW_MODE_TOGGLE)
 	else
-		throw_mode_on()
+		throw_mode_on(THROW_MODE_TOGGLE)
 
 
-/mob/living/carbon/proc/throw_mode_off()
-	in_throw_mode = FALSE
+/mob/living/carbon/proc/throw_mode_off(method)
+	if(throw_mode > method) //A toggle doesnt affect a hold
+		return
+	throw_mode = THROW_MODE_DISABLED
```

Comment:
I feel that the way we currently handle it, toggling throw on (which should be an user input), is very hacky.

How about a `can_throw_or_catch` lazylist instead, storing the sources, `throw_mode` being one of them? That's more expansible.

Reply:
I don't really see any benefits to this, throwing/catching doesn't seem like just an input, can you elaborate on this?

Reply:
Talked about it on discord, was deemed a refactor Floyd wasn't willing to do.
I personally don't like this feature as it will add code debt someone else will have to clean up after, but I understand the changes necessary to make this better are not trivial.

That said, could you make it either start undefined or defined to a less noble key than space? Seems a bit niche and specific for such a big hotkey, which can easily be fat-fingered.

Reply:
I spoke to a lot of people and they said they liked Space, which is why I used it. Fat-fingering it wouldnt matter because it doesnt do anything on tap.